### DISCUSSION TITLE
"Segmentation violation detected" with HSL solvers

### DISCUSSION THREAD

**Discussion by guigau17** [coin-or/Ipopt#448] [Q&A] [answered]
Hi,

I'm using Ipopt through CasADi for optimal control problems (Matlab R2020b, macOS 11.2.1). I have installed the additional solvers from HSL. The solvers work with a simple NLP test problem. However, when I use them for my PhD project, I get a "Segmentation violation detected" and Matlab crashes. I'm trying to understand what causes this crash..
Any help/tips would be appreciated ! Thanks a lot !

**Comment by svigerske** [comment, marked as answer]
I recall some problems with Lapack and using 32 vs 64 bit integer. Ipopt uses 32-bit int, while Matlab brings a Lapack implementation that uses 64 bit integers, so there are problems if the Lapack calls in Ipopt end up in Matlabs Lapack library.
And then there is also MA57 that is included in Matlab already and uses long integers.

**Reply by guigau17** [reply]
Thank you for your answer !
Would you know if there is a way of calling external libraries of Lapack (and Blas ?) that uses 32-bit in Matlab instead of the ones already installed ? I'm not familiar with how this works.
If not, I'll try to work with MA57 that is already included in Matlab.

**Reply by svigerske** [reply]
I'm not familiar with Matlab either. Maybe the maintainers of the [Ipopt/Matlab interface](https://github.com/ebertolazzi/mexIPOPT) know.

There is a define (`FUNNY_MA57_FINT`) one can set for the [MA57 interface](https://github.com/coin-or/Ipopt/blob/master/src/Algorithm/LinearSolvers/IpMa57TSolverInterface.hpp) that was meant for those that want to use the MA57 that comes with Matlab. I don't know if any of that still works (I don't have Matlab).

**Reply by guigau17** [reply]
I'll look into that, thank you very much !
